### ISSUE TITLE
Can no longer import Panel 1.0.4 in PyScript/Pyodide

### ISSUE THREAD

**Issue by owenlamont** [holoviz/panel#5034]
#### ALL software version info

Browser: Vivaldi 6.0.2979.22 (Stable channel) (64-bit)
Panel 1.0.4
Pyscript 2023.03.01
Bokeh 3.1.1
Tried pyodide 0.22.1 and pyodide 0.23.2

#### Description of expected behavior and the observed behavior

I was able to import Panel 1.0.4 in PyScript just yesterday but now it won't even import. It will no longer install because it apparently can't resolve a version of markdown-it-py that satisfies requirements.

Apologies if I'm raising this in the wrong place, wasn't sure if this is a Panel issue or [PyScript issue](https://github.com/pyscript/pyscript/issues).

#### Complete, minimal, self-contained example code that reproduces the issue

**index.html**
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <title>Test Dependencies</title>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <link rel="stylesheet" href="https://pyscript.net/releases/2023.03.1/pyscript.css" />
    <script
            type="text/javascript"
            src="https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js"
    ></script>
    <script
            type="text/javascript"
            src="https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.1.1.min.js"
    ></script>
    <script
            type="text/javascript"
            src="https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.1.1.min.js"
    ></script>
    <script
            type="text/javascript"
            src="https://cdn.jsdelivr.net/npm/@holoviz/panel@1.0.4/dist/panel.min.js"
    ></script>
    <script defer src="https://pyscript.net/releases/2023.03.1/pyscript.js"></script>
</head>
<body>
    <py-config src="./pyscript.toml"></py-config>
    <py-script src="./main.py"></py-script>
    <div id="plot"></div>
</body>
</html>
```

**main.py**
```python
print("Hello, World!")
```

**pyscript.toml**
```toml
name = "Test Dependencies"

packages = [
    "https://cdn.holoviz.org/panel/1.0.4/dist/wheels/bokeh-3.1.1-py3-none-any.whl",
    "https://cdn.holoviz.org/panel/1.0.4/dist/wheels/panel-1.0.4-py3-none-any.whl",
]
```

#### Stack traceback and/or browser JavaScript console output

```
(PY1001): Unable to install package(s) 'https://cdn.holoviz.org/panel/1.0.4/dist/wheels/bokeh-3.1.1-py3-none-any.whl,https://cdn.holoviz.org/panel/1.0.4/dist/wheels/panel-1.0.4-py3-none-any.whl'. Reason: Traceback (most recent call last): File "/lib/python3.10/asyncio/futures.py", line 201, in result raise self._exception File "/lib/python3.10/asyncio/tasks.py", line 234, in __step result = coro.throw(exc) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 573, in install await transaction.gather_requirements(requirements) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 333, in gather_requirements await gather(*requirement_promises) File "/lib/python3.10/asyncio/futures.py", line 284, in __await__ yield self # This tells Task to wait for completion. File "/lib/python3.10/asyncio/tasks.py", line 304, in __wakeup future.result() File "/lib/python3.10/asyncio/futures.py", line 201, in result raise self._exception File "/lib/python3.10/asyncio/tasks.py", line 234, in __step result = coro.throw(exc) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 346, in add_requirement await self.add_wheel(wheel, extras=set()) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 463, in add_wheel await self.gather_requirements(wheel.requires(extras)) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 333, in gather_requirements await gather(*requirement_promises) File "/lib/python3.10/asyncio/futures.py", line 284, in __await__ yield self # This tells Task to wait for completion. File "/lib/python3.10/asyncio/tasks.py", line 304, in __wakeup future.result() File "/lib/python3.10/asyncio/futures.py", line 201, in result raise self._exception File "/lib/python3.10/asyncio/tasks.py", line 234, in __step result = coro.throw(exc) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 337, in add_requirement return await self.add_requirement_inner(req) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 448, in add_requirement_inner await self.add_wheel(wheel, req.extras) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 463, in add_wheel await self.gather_requirements(wheel.requires(extras)) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 333, in gather_requirements await gather(*requirement_promises) File "/lib/python3.10/asyncio/futures.py", line 284, in __await__ yield self # This tells Task to wait for completion. File "/lib/python3.10/asyncio/tasks.py", line 304, in __wakeup future.result() File "/lib/python3.10/asyncio/futures.py", line 201, in result raise self._exception File "/lib/python3.10/asyncio/tasks.py", line 232, in __step result = coro.send(None) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 337, in add_requirement return await self.add_requirement_inner(req) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 418, in add_requirement_inner if self.check_version_satisfied(req): File "/lib/python3.10/site-packages/micropip/_micropip.py", line 364, in check_version_satisfied raise ValueError( ValueError: Requested 'markdown-it-py<3.0.0,>=1.0.0', but markdown-it-py==3.0.0 is already installed . Please open an issue at https://github.com/pyscript/pyscript/issues/new if you require help or you think it's a bug.
```

#### Screenshots or screencasts of the bug in action

![image](https://github.com/holoviz/panel/assets/12672027/c4b33564-9245-4838-9e0a-b331c4df1644)

**Comment by owenlamont** [comment]
I'm not sure if something changed in PyScript or the wheel files (no version numbers have changed) that I can see which is really confusing.

I tested the same code but reverting to Panel 0.14.4 and Bokeh 2.4.3 and that's still fine.

**Comment by owenlamont** [comment]
I'm guessing this is probably some dependency update that has broken things and perhaps some version needs to be pinned

**Comment by Hoxbro** [comment]
This was fixed in https://github.com/holoviz/panel/pull/5031 and should be available in the next release 1.0.5

**Comment by philippjfr** [comment]
The problem is mdit-py-plugins which has not yet released a compatible version. They've already patched it here https://github.com/executablebooks/mdit-py-plugins/commit/c297f7e2ea6420fe9c75bc493088a04df6d3118d but haven't released the new version yet. 

I also strongly dislike the fact that pyodide will not force a downgrade of an existing installed package leading to unresolvable environments like this.
